server/proxy/call.go

```diff
@@ -178,3 +180,19 @@ func (p *Proxy) getStatic(r *incoming.Request, app apps.App, path string) (io.Re
 	}
 	return up.GetStatic(r.Ctx(), app, path)
 }
+
+// pingApp checks if the app is accessible. Call its ping path with nothing
+// expanded, ignore 404 errors coming back and consider everything else a
+// "success".
+func (p *Proxy) pingApp(r *incoming.Request, app apps.App) (reachable bool) {
+	_, err := p.callApp(r, app, apps.CallRequest{Call: apps.DefaultPing})
+
+	return err == nil || errors.Cause(err) == utils.ErrNotFound
+}
+
+func (p *Proxy) timeoutRequest(r *incoming.Request, timeout time.Duration) (*incoming.Request, context.CancelFunc) {
+	r = r.Clone()
+	ctx, cancel := context.WithTimeout(r.Ctx(), timeout)
+	incoming.WithCtx(ctx)(r)
+	return r, cancel
+}
```

Comment:
Should this be a method of `incoming.Request`? What does it have to do with `Proxy`?

Reply:
See https://github.com/mattermost/mattermost-plugin-apps/pull/331